### ISSUE TITLE
Avoid crashing the VSCode extension

### ISSUE THREAD

**Issue by tvvignesh** [apollographql/apollo-tooling#1874]
Hi. If there are issues with the GraphQL schema like say, 2 GQL queries using the same name, or invalid GQL file and so on, the extension reports the error, retries 3-4 times and crashes.

For eg. this is a valid error:

![Screenshot from 2020-04-06 15-20-22](https://user-images.githubusercontent.com/1165845/78555344-754a8900-782a-11ea-999d-311bf7e2979f.png)

But when issues like this occur, this happens:

![Screenshot](https://user-images.githubusercontent.com/1165845/68545541-11936f80-03f4-11ea-822c-5bf8b75d3e3d.png)

Related https://github.com/apollographql/apollo-tooling/issues/801#issuecomment-552200208

You can get it working again only after you restart VSCode. It would be great, if it shows the error but the extension does not crash so that I can fix the error but at the same time have it working for the remaining files.

Or, if that's not possible, atleast a way to restart the extension without closing the VSCode window since it is inconvenient to do that everytime. Hope this makes sense.

Thanks.

**Comment by tafelito** [comment]
Not only it would be ideal not to crash when this kind of errors happen, it would be also great to allow to configure if switching to the output should happen automatically.

**Comment by mdecurtins** [comment]
Agreed. This is decidedly sub-optimal behavior. If it's not possible to auto-restart the plugin after a crash, could there at least be a command available in the VS Code command palette that permits manually restarting the plugin?
